// @turtlemint/mint-ui/dist/src/modal/confirm.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Confirm = void 0;
var React = __importStar(require("react"));
var dialog_1 = __importDefault(require("./dialog"));
var index_1 = require("./index");
var button_1 = __importDefault(require("../button"));
var type_1 = require("../__utils/type");
var ConfirmValues = ["info", "success", "error", "warning"];
var ConfirmTuple = type_1.tuple.apply(void 0, ConfirmValues);
exports.Confirm = function (_a) {
    var container = _a.container, visible = _a.visible, title = _a.title, width = _a.width, style = _a.style, bodyStyle = _a.bodyStyle, onCancel = _a.onCancel, cancelButtonProps = _a.cancelButtonProps, cancelText = _a.cancelText, okType = _a.okType, onOk = _a.onOk, okButtonProps = _a.okButtonProps, okText = _a.okText, type = _a.type, icon = _a.icon, iconColor = _a.iconColor, children = _a.children, rest = __rest(_a, ["container", "visible", "title", "width", "style", "bodyStyle", "onCancel", "cancelButtonProps", "cancelText", "okType", "onOk", "okButtonProps", "okText", "type", "icon", "iconColor", "children"]);
    return visible ? (React.createElement(dialog_1.default, __assign({ visible: visible, width: width, container: container, style: style }, rest),
        React.createElement(index_1.Body, { style: __assign({ padding: "24px 24px 16px" }, bodyStyle), className: "tm-modal-confirm-body" },
            React.createElement(index_1.Header, { icon: icon, iconColor: iconColor, title: title, className: "tm-modal-confirm-header" }),
            React.createElement("div", { style: {
                    marginTop: "8px",
                    marginLeft: "38px"
                } }, children),
            React.createElement(index_1.Footer, { style: {
                    marginTop: "24px",
                    display: "flex",
                    justifyContent: "flex-end",
                    alignItems: "center"
                }, isConfirm: true, className: "tm-modal-confirm-footer" },
                type && ConfirmValues.includes(type) ? null : (React.createElement(button_1.default, __assign({ title: cancelText ? cancelText : "Cancel", style: { marginRight: "10px" }, onClick: onCancel }, cancelButtonProps))),
                React.createElement(button_1.default, __assign({ title: okText ? okText : "OK", btnType: okType ? okType : "solid", btnStyle: "primary", onClick: onOk }, okButtonProps)))))) : null;
};
exports.default = exports.Confirm;
