// @turtlemint/mint-ui/dist/src/collapse/index.js
"use strict";
var __makeTemplateObject = (this && this.__makeTemplateObject) || function (cooked, raw) {
    if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
    return cooked;
};
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Header = exports.Panel = exports.Collapse = void 0;
var React = __importStar(require("react"));
var styled_components_1 = __importStar(require("styled-components"));
var polished_1 = require("polished");
var colors_1 = __importDefault(require("../__utils/colors"));
var KeyboardArrowDown_1 = __importDefault(require("../icons/KeyboardArrowDown"));
var KeyboardArrowRight_1 = __importDefault(require("../icons/KeyboardArrowRight"));
var app_1 = require("../app");
var typography_1 = require("../typography");
var CollapseContext = React.createContext({});
exports.Collapse = function (_a) {
    var _b = _a.activeKey, activeKey = _b === void 0 ? 1 : _b, onChange = _a.onChange, panels = _a.panels, expandIconPosition = _a.expandIconPosition;
    return (React.createElement(CollapseContext.Provider, { value: {
            activeKey: activeKey,
            onChange: onChange,
            expandIconPosition: expandIconPosition
        } },
        React.createElement(CollapseWrap, null, panels.map(function (panel) { return (React.createElement(exports.Panel, __assign({ key: panel.panelKey }, panel))); }))));
};
var PanelContext = React.createContext({});
exports.Panel = function (_a) {
    var panelKey = _a.panelKey, header = _a.header, content = _a.content, _b = _a.disabled, disabled = _b === void 0 ? false : _b;
    var activeKey = React.useContext(CollapseContext).activeKey;
    var open = activeKey === panelKey;
    return (React.createElement(PanelContext.Provider, { value: {
            panelKey: panelKey,
            disabled: disabled
        } },
        React.createElement("div", { className: "tm-panel" },
            open ? null : React.createElement(exports.Header, __assign({}, header, { open: open })),
            open ? (React.createElement(Body, { className: "tm-panel-body" },
                React.createElement(typography_1.Title, { text: header.text, level: 4, weight: "medium" }),
                React.createElement("div", { style: {
                        marginTop: "30px"
                    } }, content))) : null)));
};
exports.Header = function (_a) {
    var text = _a.text, style = _a.style, extra = _a.extra, open = _a.open;
    var _b = React.useContext(CollapseContext), onChange = _b.onChange, expandIconPosition = _b.expandIconPosition;
    var _c = React.useContext(PanelContext), panelKey = _c.panelKey, disabled = _c.disabled;
    var getIcon = function () {
        return open ? (React.createElement(Icon, { position: expandIconPosition },
            React.createElement(KeyboardArrowDown_1.default, { width: 24, height: 24, fill: colors_1.default.GREY3 }))) : (React.createElement(Icon, { position: expandIconPosition },
            React.createElement(KeyboardArrowRight_1.default, { width: 24, height: 24, fill: colors_1.default.GREY3 })));
    };
    return (React.createElement(StyledHeader, { disabled: disabled, style: style, onClick: function () { return onChange(panelKey); }, className: "tm-panel-header" },
        React.createElement("div", { style: { display: "inline-flex" } },
            expandIconPosition === "left" ? getIcon() : null,
            React.createElement(typography_1.Title, { text: text, level: 4, weight: "medium" })),
        React.createElement("div", { style: { display: "inline-flex" } },
            extra ? extra : null,
            expandIconPosition === "right" ? getIcon() : null)));
};
var CollapseWrap = styled_components_1.default.div(templateObject_1 || (templateObject_1 = __makeTemplateObject(["\n\t", ";\n\tborder-left: 1px solid ", ";\n\tborder-right: 1px solid ", ";\n\tborder-radius: 4px;\n\t.tm-panel:first-of-type .tm-panel-header {\n\t\tborder-top: 1px solid ", ";\n\t}\n\t.tm-panel:first-of-type .tm-panel-body {\n\t\tborder-top: 1px solid ", ";\n\t}\n"], ["\n\t", ";\n\tborder-left: 1px solid ", ";\n\tborder-right: 1px solid ", ";\n\tborder-radius: 4px;\n\t.tm-panel:first-of-type .tm-panel-header {\n\t\tborder-top: 1px solid ", ";\n\t}\n\t.tm-panel:first-of-type .tm-panel-body {\n\t\tborder-top: 1px solid ", ";\n\t}\n"])), app_1.GlobalStyles, colors_1.default.BORDER, colors_1.default.BORDER, colors_1.default.BORDER, colors_1.default.BORDER);
var StyledHeader = styled_components_1.default.div(templateObject_3 || (templateObject_3 = __makeTemplateObject(["\n\tdisplay: flex;\n\talign-items: center;\n\tjustify-content: space-between;\n\tpadding: 12px 45px;\n\tcolor: ", ";\n\tbackground: ", ";\n\tborder-bottom: 1px solid ", ";\n\tcursor: pointer;\n\t", ";\n"], ["\n\tdisplay: flex;\n\talign-items: center;\n\tjustify-content: space-between;\n\tpadding: 12px 45px;\n\tcolor: ", ";\n\tbackground: ", ";\n\tborder-bottom: 1px solid ", ";\n\tcursor: pointer;\n\t",
    ";\n"])), colors_1.default.INACTIVE_HEADING, colors_1.default.BACKGROUND_COLOR_LIGHT, colors_1.default.BORDER, function (_a) {
    var disabled = _a.disabled;
    return disabled && styled_components_1.css(templateObject_2 || (templateObject_2 = __makeTemplateObject(["\n\t\t\tpointer-events: none;\n\t\t\tcolor: ", ";\n\t\t\tbackground: ", ";\n\t\t"], ["\n\t\t\tpointer-events: none;\n\t\t\tcolor: ", ";\n\t\t\tbackground: ", ";\n\t\t"])), polished_1.transparentize(0.7, colors_1.default.BLACK), polished_1.transparentize(0.2, colors_1.default.BACKGROUND_GREY));
});
var Body = styled_components_1.default.div(templateObject_4 || (templateObject_4 = __makeTemplateObject(["\n\tpadding: 12px 45px;\n\tborder-bottom: 1px solid ", ";\n\tbackground: ", ";\n"], ["\n\tpadding: 12px 45px;\n\tborder-bottom: 1px solid ", ";\n\tbackground: ", ";\n"])), colors_1.default.BORDER, colors_1.default.WHITE);
var Icon = function (_a) {
    var position = _a.position, children = _a.children;
    var StyledEL = styled_components_1.default.i(templateObject_6 || (templateObject_6 = __makeTemplateObject(["\n\t\tdisplay: inline-flex;\n\t\tmargin-left: 10px;\n\t\t", ";\n\t"], ["\n\t\tdisplay: inline-flex;\n\t\tmargin-left: 10px;\n\t\t",
        ";\n\t"])), function (_a) {
        var position = _a.position;
        return position === "left" && styled_components_1.css(templateObject_5 || (templateObject_5 = __makeTemplateObject(["\n\t\t\t\tmargin: 0;\n\t\t\t\tmargin-right: 10px;\n\t\t\t"], ["\n\t\t\t\tmargin: 0;\n\t\t\t\tmargin-right: 10px;\n\t\t\t"])));
    });
    return React.createElement(StyledEL, { position: position }, children);
};
exports.default = exports.Collapse;
var templateObject_1, templateObject_2, templateObject_3, templateObject_4, templateObject_5, templateObject_6;
