// @turtlemint/mint-ui/dist/src/input/index.js
"use strict";
var __makeTemplateObject = (this && this.__makeTemplateObject) || function (cooked, raw) {
    if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
    return cooked;
};
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __spreadArrays = (this && this.__spreadArrays) || function () {
    for (var s = 0, i = 0, il = arguments.length; i < il; i++) s += arguments[i].length;
    for (var r = Array(s), k = 0, i = 0; i < il; i++)
        for (var a = arguments[i], j = 0, jl = a.length; j < jl; j++, k++)
            r[k] = a[j];
    return r;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.StyledInput = exports.Input = exports.DEFAULT_VALIDATION = void 0;
var React = __importStar(require("react"));
var styled_components_1 = __importStar(require("styled-components"));
var colors_1 = __importDefault(require("../__utils/colors"));
var app_1 = require("../app");
var validators_1 = require("@turtlemint/validators");
var typography_1 = require("../typography");
var icon_1 = require("../icon");
// Default validaiton for `ALPHA NUMBERIC	`
exports.DEFAULT_VALIDATION = ["alphaNumericWithSpace"];
exports.Input = React.forwardRef(function (_a, ref) {
    var _b = _a.type, type = _b === void 0 ? "text" : _b, _c = _a.name, name = _c === void 0 ? "" : _c, placeholder = _a.placeholder, _d = _a.error, error = _d === void 0 ? true : _d, _e = _a.block, block = _e === void 0 ? false : _e, _f = _a.disabled, disabled = _f === void 0 ? false : _f, _g = _a.onChange, onChange = _g === void 0 ? function () { } : _g, _h = _a.onBlur, onBlur = _h === void 0 ? function () { } : _h, value = _a.value, _j = _a.className, className = _j === void 0 ? "" : _j, size = _a.size, _k = _a.validations, validations = _k === void 0 ? [] : _k, _l = _a.validateOnBlur, validateOnBlur = _l === void 0 ? false : _l, _m = _a.skipDefaultValidation, skipDefaultValidation = _m === void 0 ? false : _m, iconStyle = _a.iconStyle, icon = _a.icon, iconPosition = _a.iconPosition, iconColor = _a.iconColor, _o = _a.showErrorMessage, showErrorMessage = _o === void 0 ? true : _o, _p = _a.theme, theme = _p === void 0 ? colors_1.default : _p, rest = __rest(_a, ["type", "name", "placeholder", "error", "block", "disabled", "onChange", "onBlur", "value", "className", "size", "validations", "validateOnBlur", "skipDefaultValidation", "iconStyle", "icon", "iconPosition", "iconColor", "showErrorMessage", "theme"]);
    var finalTheme = __assign(__assign({}, colors_1.default), theme);
    var textRef = React.useRef();
    // Final set of validtions to perform
    var finalValidations = __spreadArrays(validations, (!skipDefaultValidation ? exports.DEFAULT_VALIDATION : []));
    var _q = React.useState({
        isValid: true,
        message: ""
    }), validationState = _q[0], updateValidationState = _q[1];
    var handleChange = function (e) {
        !validateOnBlur && handleValidation(e.target.value);
        onChange(e.target.value, e.target.name);
    };
    var handleBlur = function (e) {
        validateOnBlur && handleValidation(e.target.value);
        onBlur(e.target.value);
    };
    var handleValidation = function (updatedValue) {
        var validationObj = validators_1.validationHandler(updatedValue, finalValidations);
        updateValidationState(__assign({}, validationObj));
    };
    var isValid = function (withMessage) {
        if (withMessage === void 0) { withMessage = false; }
        var currentValue = textRef.current.value;
        var validationObj = validators_1.validationHandler(currentValue, finalValidations);
        updateValidationState(__assign({}, validationObj));
        return withMessage ? validationObj : validationObj.isValid;
    };
    // Exposed public function for validation check
    React.useImperativeHandle(ref, function () { return ({
        isValid: isValid
    }); });
    return (React.createElement(StyledWrapper, { block: block },
        icon ? (React.createElement(IconWrapper, { inputSize: size, iconPosition: iconPosition, style: iconStyle, theme: finalTheme },
            React.createElement(icon_1.Icon, __assign({ name: icon, color: iconColor }, rest.iconProps)))) : null,
        React.createElement(exports.StyledInput, __assign({ ref: textRef, name: name, type: type, placeholder: placeholder, error: error ? !validationState.isValid : false, disabled: disabled, onBlur: handleBlur, onChange: handleChange, value: value, className: className, "aria-label": "tm-input", inputSize: size }, rest, { theme: finalTheme, iconPosition: iconPosition })),
        error && showErrorMessage && (React.createElement(typography_1.Text, { textType: "danger", text: validationState.message }))));
});
var StyledWrapper = styled_components_1.default.div(templateObject_1 || (templateObject_1 = __makeTemplateObject(["\n\t", ";\n\twidth: ", ";\n\ttext-align: left;\n\tposition: relative;\n"], ["\n\t", ";\n\twidth: ", ";\n\ttext-align: left;\n\tposition: relative;\n"])), app_1.GlobalStyles, function (_a) {
    var block = _a.block;
    return (block ? "100%" : "328px");
});
var IconWrapper = styled_components_1.default.div(templateObject_8 || (templateObject_8 = __makeTemplateObject(["\nposition:absolute;\nheight:96%;\n// background-color:", ";\ndisplay:flex;\nalign-items:center;\njustify-content:center;\npadding-left:12px;\npadding-right:12px;\n// border: 1px solid ", ";\n", "\n\t", "\n\n\n", "\n", "\n", "\n\t", "\n"], ["\nposition:absolute;\nheight:96%;\n// background-color:", ";\ndisplay:flex;\nalign-items:center;\njustify-content:center;\npadding-left:12px;\npadding-right:12px;\n// border: 1px solid ", ";\n",
    "\n\t",
    "\n\n\n",
    "\n",
    "\n",
    "\n\t",
    "\n"])), function (_a) {
    var theme = _a.theme;
    return theme.WHITE;
}, function (_a) {
    var theme = _a.theme;
    return theme.GREY4;
}, function (_a) {
    var iconPosition = _a.iconPosition;
    return iconPosition === "right" && styled_components_1.css(templateObject_2 || (templateObject_2 = __makeTemplateObject(["\n\t\tright: 0px;\n\t\tborder-left: none;\n\t"], ["\n\t\tright: 0px;\n\t\tborder-left: none;\n\t"])));
}, function (_a) {
    var iconPosition = _a.iconPosition;
    return iconPosition === "left" && styled_components_1.css(templateObject_3 || (templateObject_3 = __makeTemplateObject(["\n\t\t\tleft: 0px;\n\t\t\tborder-right: none;\n\t\t"], ["\n\t\t\tleft: 0px;\n\t\t\tborder-right: none;\n\t\t"])));
}, function (_a) {
    var iconPosition = _a.iconPosition, inputSize = _a.inputSize;
    return iconPosition === "left" &&
        inputSize === "small" && styled_components_1.css(templateObject_4 || (templateObject_4 = __makeTemplateObject(["\n\t\tleft: 0px;\n\t\tpadding-left: 8px;\n\t\tpadding-right: 8px;\n\t\tborder-right: none;\n\t"], ["\n\t\tleft: 0px;\n\t\tpadding-left: 8px;\n\t\tpadding-right: 8px;\n\t\tborder-right: none;\n\t"])));
}, function (_a) {
    var iconPosition = _a.iconPosition, inputSize = _a.inputSize;
    return iconPosition === "left" &&
        inputSize === "large" && styled_components_1.css(templateObject_5 || (templateObject_5 = __makeTemplateObject(["\n\t\tleft: 0px;\n\t\tpadding-left: 14px;\n\t\tpadding-right: 14px;\n\t\tborder-right: none;\n\t"], ["\n\t\tleft: 0px;\n\t\tpadding-left: 14px;\n\t\tpadding-right: 14px;\n\t\tborder-right: none;\n\t"])));
}, function (_a) {
    var iconPosition = _a.iconPosition, inputSize = _a.inputSize;
    return iconPosition === "right" &&
        inputSize === "small" && styled_components_1.css(templateObject_6 || (templateObject_6 = __makeTemplateObject(["\n\t\tright: 0px;\n\t\tpadding-left: 8px;\n\t\tpadding-right: 8px;\n\t\tborder-left: none;\n\t"], ["\n\t\tright: 0px;\n\t\tpadding-left: 8px;\n\t\tpadding-right: 8px;\n\t\tborder-left: none;\n\t"])));
}, function (_a) {
    var iconPosition = _a.iconPosition, inputSize = _a.inputSize;
    return iconPosition === "right" &&
        inputSize === "large" && styled_components_1.css(templateObject_7 || (templateObject_7 = __makeTemplateObject(["\n\t\t\tright: 0px;\n\t\t\tpadding-left: 14px;\n\t\t\tpadding-right: 14px;\n\t\t\tborder-left: none;\n\t\t"], ["\n\t\t\tright: 0px;\n\t\t\tpadding-left: 14px;\n\t\t\tpadding-right: 14px;\n\t\t\tborder-left: none;\n\t\t"])));
});
exports.StyledInput = styled_components_1.default.input(templateObject_15 || (templateObject_15 = __makeTemplateObject(["\n\tbox-sizing: border-box;\n\tborder: 0;\n\tborder: 1px solid ", ";\n\tpadding: 12px 48px 12px 16px;\n\tborder-radius: 4px;\n\tcolor: ", ";\n\twidth: 100%;\n\tfont-size: 14px;\n\toutline-color: ", ";\n\t::-webkit-input-placeholder {\n\t\t/* Chrome/Opera/Safari */\n\t\tcolor: ", ";\n\t}\n\t::-moz-placeholder {\n\t\t/* Firefox 19+ */\n\t\tcolor: ", ";\n\t}\n\t:-ms-input-placeholder {\n\t\t/* IE 10+ */\n\t\tcolor: ", ";\n\t}\n\t:-moz-placeholder {\n\t\t/* Firefox 18- */\n\t\tcolor: ", ";\n\t}\n\t::placeholder: {\n\t\tcolor: ", ";\n\t}\n\t", "\n", "\n\n\t", ";\n\t", ";\n\t", "\n\t", "\n"], ["\n\tbox-sizing: border-box;\n\tborder: 0;\n\tborder: 1px solid ", ";\n\tpadding: 12px 48px 12px 16px;\n\tborder-radius: 4px;\n\tcolor: ", ";\n\twidth: 100%;\n\tfont-size: 14px;\n\toutline-color: ", ";\n\t::-webkit-input-placeholder {\n\t\t/* Chrome/Opera/Safari */\n\t\tcolor: ", ";\n\t}\n\t::-moz-placeholder {\n\t\t/* Firefox 19+ */\n\t\tcolor: ", ";\n\t}\n\t:-ms-input-placeholder {\n\t\t/* IE 10+ */\n\t\tcolor: ", ";\n\t}\n\t:-moz-placeholder {\n\t\t/* Firefox 18- */\n\t\tcolor: ", ";\n\t}\n\t::placeholder: {\n\t\tcolor: ", ";\n\t}\n\t",
    "\n",
    "\n\n\t",
    ";\n\t",
    ";\n\t",
    "\n\t",
    "\n"])), function (_a) {
    var theme = _a.theme;
    return theme.GREY4;
}, function (_a) {
    var theme = _a.theme;
    return theme.GREY1;
}, function (_a) {
    var theme = _a.theme;
    return theme.PRIMARY_LIGHT;
}, function (_a) {
    var theme = _a.theme;
    return theme.DISABLED;
}, function (_a) {
    var theme = _a.theme;
    return theme.DISABLED;
}, function (_a) {
    var theme = _a.theme;
    return theme.DISABLED;
}, function (_a) {
    var theme = _a.theme;
    return theme.DISABLED;
}, function (_a) {
    var theme = _a.theme;
    return theme.DISABLED;
}, function (_a) {
    var iconPosition = _a.iconPosition;
    return iconPosition === "right" && styled_components_1.css(templateObject_9 || (templateObject_9 = __makeTemplateObject(["\n\t\t\tpadding-right: 48px;\n\t\t"], ["\n\t\t\tpadding-right: 48px;\n\t\t"])));
}, function (_a) {
    var iconPosition = _a.iconPosition;
    return iconPosition === "left" && styled_components_1.css(templateObject_10 || (templateObject_10 = __makeTemplateObject(["\n\t\tpadding-left: 48px;\n\t"], ["\n\t\tpadding-left: 48px;\n\t"])));
}, function (props) {
    return props.error && styled_components_1.css(templateObject_11 || (templateObject_11 = __makeTemplateObject(["\n\t\t\tborder-color: ", ";\n\t\t\t&:focus {\n\t\t\t\toutline-color: ", ";\n\t\t\t}\n\t\t"], ["\n\t\t\tborder-color: ", ";\n\t\t\t&:focus {\n\t\t\t\toutline-color: ", ";\n\t\t\t}\n\t\t"])), props.theme.DANGER, props.theme.DANGER);
}, function (props) {
    return props.disabled && styled_components_1.css(templateObject_12 || (templateObject_12 = __makeTemplateObject(["\n\t\t\topacity: 0.64;\n\t\t\tcursor: no-drop;\n\t\t\tbackground-color: ", ";\n\t\t\tcolor: ", ";\n\t\t\t::-webkit-input-placeholder {\n\t\t\t\t/* Chrome/Opera/Safari */\n\t\t\t\tcolor: ", ";\n\t\t\t}\n\t\t\t::-moz-placeholder {\n\t\t\t\t/* Firefox 19+ */\n\t\t\t\tcolor: ", ";\n\t\t\t}\n\t\t\t:-ms-input-placeholder {\n\t\t\t\t/* IE 10+ */\n\t\t\t\tcolor: ", ";\n\t\t\t}\n\t\t\t:-moz-placeholder {\n\t\t\t\t/* Firefox 18- */\n\t\t\t\tcolor: ", ";\n\t\t\t}\n\t\t\t::placeholder: {\n\t\t\t\tcolor: ", ";\n\t\t\t}\n\t\t"], ["\n\t\t\topacity: 0.64;\n\t\t\tcursor: no-drop;\n\t\t\tbackground-color: ", ";\n\t\t\tcolor: ", ";\n\t\t\t::-webkit-input-placeholder {\n\t\t\t\t/* Chrome/Opera/Safari */\n\t\t\t\tcolor: ", ";\n\t\t\t}\n\t\t\t::-moz-placeholder {\n\t\t\t\t/* Firefox 19+ */\n\t\t\t\tcolor: ", ";\n\t\t\t}\n\t\t\t:-ms-input-placeholder {\n\t\t\t\t/* IE 10+ */\n\t\t\t\tcolor: ", ";\n\t\t\t}\n\t\t\t:-moz-placeholder {\n\t\t\t\t/* Firefox 18- */\n\t\t\t\tcolor: ", ";\n\t\t\t}\n\t\t\t::placeholder: {\n\t\t\t\tcolor: ", ";\n\t\t\t}\n\t\t"])), props.theme.DISABLED, props.theme.GREY3, props.theme.GREY3, props.theme.GREY3, props.theme.GREY3, props.theme.GREY3, props.theme.GREY3);
}, function (_a) {
    var inputSize = _a.inputSize;
    return inputSize === "small" && styled_components_1.css(templateObject_13 || (templateObject_13 = __makeTemplateObject(["\n\t\t\tpadding-top: 8px;\n\t\t\tpadding-bottom: 8px;\n\t\t"], ["\n\t\t\tpadding-top: 8px;\n\t\t\tpadding-bottom: 8px;\n\t\t"])));
}, function (_a) {
    var inputSize = _a.inputSize;
    return inputSize === "large" && styled_components_1.css(templateObject_14 || (templateObject_14 = __makeTemplateObject(["\n\t\t\tpadding-top: 14px;\n\t\t\tpadding-bottom: 14px;\n\t\t"], ["\n\t\t\tpadding-top: 14px;\n\t\t\tpadding-bottom: 14px;\n\t\t"])));
});
// Issue = https://github.com/DefinitelyTyped/DefinitelyTyped/issues/37660
// Input.propTypes = {
// 	type: PropTypes.string,
// 	name: PropTypes.string,
// 	placeholder: PropTypes.string,
// 	error: PropTypes.string,
// 	block: PropTypes.bool,
// 	disabled: PropTypes.bool,
// 	onChange: PropTypes.func,
// 	onBlur: PropTypes.func,
// 	value: PropTypes.string,
// 	className: PropTypes.string,
// 	size: PropTypes.oneOf(["large", "small", "default"])
// };
exports.Input.defaultProps = {
    type: "text",
    block: false,
    disabled: false
};
exports.default = React.memo(styled_components_1.withTheme(exports.Input));
var templateObject_1, templateObject_2, templateObject_3, templateObject_4, templateObject_5, templateObject_6, templateObject_7, templateObject_8, templateObject_9, templateObject_10, templateObject_11, templateObject_12, templateObject_13, templateObject_14, templateObject_15;
