// @turtlemint/mint-ui/dist/src/__utils/colors.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.COLORS = void 0;
exports.COLORS = {
    PRIMARY: "#00a465",
    PRIMARY_LIGHT: "#0fb877",
    PRIMARY_DARK: "#078a58",
    BACKGROUND_COLOR: "#f0f8fe",
    BACKGROUND_COLOR_LIGHT: "#F9FCFF",
    BACKGROUND_GREEN: "#f0fef9",
    BACKGROUND_GREEN_DARK: "#e1f5ee",
    DANGER: "#e94220",
    DANGER_LIGHT: "#f24d2c",
    DANGER_DARK: "#d43c1d",
    DISABLED: "#d2d5d6",
    WHITE: "#fff",
    BLACK: "#121617",
    GREY1: "#4d5556",
    GREY2: "#6c7375",
    GREY3: "#a1a6a7",
    GREY4: "#e8e8e8",
    GREY5: "#757575",
    BORDER: "#d9d9d9",
    INACTIVE_HEADING: "#354052",
    PERSIAN_GREEN: "#009ca4",
    BACKGROUND_GREY: "#fafafa",
    BACKGROUND_GRAY: "#f5f5f5",
    BACKGROUND_BLUE: "#d5ecfd",
    BACKGROUND_YELLOW: "#fdefce",
    BACKGROUND_MAT_YELLOW: "#E5CB76",
    BACKGROUND_BLUE_DARK: "#d3f0fb",
    BACKGROUND_RED: "#fbd9d2",
    YELLOW: "#f5ae0a",
    PICTON_BLUE: "#23b2eb",
    LINK: "#0268ed",
    HEADER_BACKGROUND: "#7f8fa4"
};
exports.default = exports.COLORS;
