// @turtlemint/mint-ui/dist/src/amount/index.js
"use strict";
var __makeTemplateObject = (this && this.__makeTemplateObject) || function (cooked, raw) {
    if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
    return cooked;
};
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __spreadArrays = (this && this.__spreadArrays) || function () {
    for (var s = 0, i = 0, il = arguments.length; i < il; i++) s += arguments[i].length;
    for (var r = Array(s), k = 0, i = 0; i < il; i++)
        for (var a = arguments[i], j = 0, jl = a.length; j < jl; j++, k++)
            r[k] = a[j];
    return r;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Amount = void 0;
var React = __importStar(require("react"));
var input_1 = __importDefault(require("../input"));
var icon_1 = __importDefault(require("../icon"));
var styled_components_1 = __importStar(require("styled-components"));
var colors_1 = __importDefault(require("../__utils/colors"));
var typography_1 = __importDefault(require("../typography"));
exports.Amount = React.forwardRef(function (_a, ref) {
    var name = _a.name, onChange = _a.onChange, onBlur = _a.onBlur, disabled = _a.disabled, style = _a.style, min = _a.min, max = _a.max, _b = _a.showConverter, showConverter = _b === void 0 ? true : _b, _c = _a.validations, validations = _c === void 0 ? [] : _c, _d = _a.error, error = _d === void 0 ? true : _d, _e = _a.value, value = _e === void 0 ? "" : _e, _f = _a.placeholder, placeholder = _f === void 0 ? "" : _f, _g = _a.theme, theme = _g === void 0 ? colors_1.default : _g, currency = _a.currency, showCurrencyAsText = _a.showCurrencyAsText;
    // default initializations
    var finalTheme = __assign(__assign({}, colors_1.default), theme);
    var _h = React.useState(value.toString()), amt = _h[0], setAmt = _h[1];
    var inputRef = React.useRef("amount");
    var convertAmt = function (number) {
        if (isNaN(number)) {
            return;
        }
        if (number < 1000) {
            return number;
        }
        else if (number < 100000) {
            return getSortedNum(number, 1000) + " k";
        }
        else if (number < 10000000) {
            return getSortedNum(number, 100000) + " Lakhs";
        }
        else if (number >= 10000000) {
            return getSortedNum(number, 10000000) + " Crs";
        }
        return number;
    };
    var getSortedNum = function (number, divider) {
        return parseFloat((number / divider).toFixed(2));
    };
    var onAmountChange = function (val) {
        setAmt(val);
        onChange && onChange(val, name || "");
    };
    // Exposed public function for validation check
    React.useImperativeHandle(ref, function () { return ({
        isValid: function () { return inputRef.current.isValid(); }
    }); });
    // Validations
    var validationArray = [{ name: "integer" }];
    // Min value validation
    if (min) {
        validationArray.push({
            name: "minValue",
            minValue: min,
            message: "Amount less than \u20B9 " + convertAmt(min) + " is not allowed"
        });
    }
    // Max value validation
    if (max) {
        validationArray.push({
            name: "maxValue",
            maxValue: max,
            message: "Amount more than \u20B9 " + convertAmt(max) + " is not allowed"
        });
    }
    if (validations) {
        validationArray = __spreadArrays(validationArray, validations);
    }
    return (React.createElement(AmtContainer, { style: style },
        React.createElement(IconTextContainer, { showCurrencyAsText: showCurrencyAsText },
            !showCurrencyAsText ? (React.createElement(icon_1.default, { name: currency ? currency : "rupee", color: colors_1.default.GREY2, size: 16 })) : null,
            showCurrencyAsText ? (React.createElement(typography_1.default, { textStyle: "uppercase", text: currency ? currency : "rupee", color: colors_1.default.GREY2, style: {
                    fontWeight: 600
                }, size: 14 })) : null),
        React.createElement(input_1.default, { value: amt, type: "number", disabled: disabled, onChange: onAmountChange, onBlur: function (selectedValue) {
                onBlur ? onBlur(selectedValue) : null;
            }, skipDefaultValidation: true, validations: validationArray, style: {
                paddingLeft: showCurrencyAsText ? "60px" : "28px"
            }, ref: inputRef, error: error, placeholder: placeholder, block: true, theme: finalTheme }),
        showConverter && convertAmt(amt) ? (React.createElement(Amt, { theme: finalTheme }, convertAmt(amt))) : null));
});
var Amt = styled_components_1.default.span(templateObject_1 || (templateObject_1 = __makeTemplateObject(["\n\tdisplay: inline-flex;\n\talign-items: center;\n\tposition: absolute;\n\tright: 2px;\n\ttop: 2px;\n\tbackground-color: ", ";\n\tpadding: 0 7px;\n\tborder-radius: 2px;\n\tfont-size: 14px;\n\theight: 38px;\n"], ["\n\tdisplay: inline-flex;\n\talign-items: center;\n\tposition: absolute;\n\tright: 2px;\n\ttop: 2px;\n\tbackground-color: ", ";\n\tpadding: 0 7px;\n\tborder-radius: 2px;\n\tfont-size: 14px;\n\theight: 38px;\n"])), function (_a) {
    var theme = _a.theme;
    return theme.GREY4;
});
var AmtContainer = styled_components_1.default.div(templateObject_2 || (templateObject_2 = __makeTemplateObject(["\n\tdisplay: flex;\n\tjustify-content: center;\n\talign-items: center;\n\tposition: relative;\n"], ["\n\tdisplay: flex;\n\tjustify-content: center;\n\talign-items: center;\n\tposition: relative;\n"])));
var IconTextContainer = styled_components_1.default.span(templateObject_3 || (templateObject_3 = __makeTemplateObject(["\n\tposition: absolute;\n\tleft: 6px;\n\tz-index: 1;\n\theight: 42px;\n\tdisplay: flex;\n\talign-items: center;\n\tjustify-content: center;\n\ttop: ", "px;\n"], ["\n\tposition: absolute;\n\tleft: 6px;\n\tz-index: 1;\n\theight: 42px;\n\tdisplay: flex;\n\talign-items: center;\n\tjustify-content: center;\n\ttop: ", "px;\n"])), function (_a) {
    var showCurrencyAsText = _a.showCurrencyAsText;
    return (showCurrencyAsText ? 0 : 2);
});
exports.default = styled_components_1.withTheme(exports.Amount);
var templateObject_1, templateObject_2, templateObject_3;
