// @turtlemint/mint-ui/dist/src/typeahead/typeahead.spec.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var React = __importStar(require("react"));
var react_1 = require("@testing-library/react");
var index_1 = __importDefault(require("./index"));
var Component = function () {
    var dataConfig = {
        url: "https://api.publicapis.org/entries",
        params: {
            // fill the params here or later in app flow dynamically as you wish
            category: "business",
            https: true
        },
        dataKey: "entries",
        valueKey: "Link",
        displayKey: "API",
        searchKey: ""
    };
    return (React.createElement("div", { style: { height: "150px", width: "300px" } },
        React.createElement(index_1.default, { name: "default-typeahead", onChange: function () { }, data: dataConfig })));
};
describe("Typeahead tests", function () {
    test("renders correctly", function () {
        var asFragment = react_1.render(React.createElement(Component, null)).asFragment;
        expect(asFragment()).toMatchSnapshot();
    });
});
