// @turtlemint/mint-ui/dist/src/table/index.js
"use strict";
var __makeTemplateObject = (this && this.__makeTemplateObject) || function (cooked, raw) {
    if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
    return cooked;
};
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __spreadArrays = (this && this.__spreadArrays) || function () {
    for (var s = 0, i = 0, il = arguments.length; i < il; i++) s += arguments[i].length;
    for (var r = Array(s), k = 0, i = 0; i < il; i++)
        for (var a = arguments[i], j = 0, jl = a.length; j < jl; j++, k++)
            r[k] = a[j];
    return r;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Table = void 0;
var React = __importStar(require("react"));
var styled_components_1 = __importDefault(require("styled-components"));
var app_1 = require("../app");
var table_head_1 = __importDefault(require("./table-head"));
var table_body_1 = __importDefault(require("./table-body"));
var pagination_1 = __importDefault(require("../pagination"));
var loading_mask_1 = __importDefault(require("../loading/loading-mask"));
var col_1 = __importDefault(require("../grid/col"));
var row_1 = __importDefault(require("../grid/row"));
exports.Table = function (_a) {
    var _b = _a.dataSource, dataSource = _b === void 0 ? [] : _b, columns = _a.columns, onRow = _a.onRow, pagination = _a.pagination, loading = _a.loading, onChange = _a.onChange;
    var _c = React.useState(dataSource), data = _c[0], setData = _c[1];
    var defaultCol = columns.filter(function (column) { return column.defaultSortOrder; })[0];
    var _d = React.useState({
        column: defaultCol,
        order: "ascends"
    }), activeSort = _d[0], setActiveSort = _d[1];
    var _e = React.useState(pagination && typeof pagination !== "boolean"
        ? {
            currentPage: pagination.currentPage,
            pageSize: pagination.pageSize
        }
        : { currentPage: 1, pageSize: 10 }), activePagination = _e[0], setActivePagination = _e[1];
    React.useEffect(function () {
        setData(dataSource);
    }, [dataSource]);
    React.useEffect(function () {
        onChange ? onChange(activePagination, activeSort) : null;
    }, [activePagination, activeSort]);
    React.useEffect(function () {
        handleSort(activeSort.column, activeSort.column.defaultSortOrder);
    }, []);
    var handleSort = function (column, sortOrder) {
        if (sortOrder === void 0) { sortOrder = "ascends"; }
        var sortedData = data.sort(function (a, b) {
            if (column && column.sorter) {
                return column.sorter(a, b, sortOrder);
            }
            return column;
        });
        setData(__spreadArrays(sortedData));
        setActiveSort({ column: column, order: sortOrder });
    };
    return (React.createElement(React.Fragment, null,
        React.createElement(TableEl, null,
            loading ? React.createElement(loading_mask_1.default, null) : null,
            React.createElement(table_head_1.default, { columns: columns, activeSort: activeSort, handleSort: handleSort }),
            React.createElement(table_body_1.default, { data: data, onRow: onRow, pageSize: activePagination.pageSize, currentPage: activePagination.currentPage, isAjax: pagination ? true : false })),
        data.length ? (React.createElement(row_1.default, { style: { marginTop: "20px" } },
            React.createElement(col_1.default, null, !pagination ? null : (React.createElement(pagination_1.default, { total: typeof pagination !== "boolean"
                    ? pagination.total
                    : dataSource.length, current: activePagination.currentPage, pageSize: activePagination.pageSize, onChange: function (paginationObj) {
                    setActivePagination(paginationObj);
                }, style: {
                    display: "flex",
                    justifyContent: "flex-end"
                } }))))) : null));
};
var TableEl = styled_components_1.default.table(templateObject_1 || (templateObject_1 = __makeTemplateObject(["\n\t", ";\n\tborder-collapse: separate;\n\tborder-spacing: 0;\n\twidth: 100%;\n\tposition: relative;\n"], ["\n\t", ";\n\tborder-collapse: separate;\n\tborder-spacing: 0;\n\twidth: 100%;\n\tposition: relative;\n"])), app_1.GlobalStyles);
exports.default = exports.Table;
var templateObject_1;
