// @turtlemint/mint-ui/dist/src/button/index.js
"use strict";
var __makeTemplateObject = (this && this.__makeTemplateObject) || function (cooked, raw) {
    if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
    return cooked;
};
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Link = exports.ButtonDefault = exports.Button = void 0;
var React = __importStar(require("react"));
var prop_types_1 = __importDefault(require("prop-types"));
var styled_components_1 = __importStar(require("styled-components"));
var type_1 = require("../__utils/type");
var colors_1 = __importDefault(require("../__utils/colors"));
var polished_1 = require("polished");
var app_1 = require("../app");
var icon_1 = __importDefault(require("../icon"));
var ButtonTypeTuple = type_1.tuple("outlined", "solid", "link");
var ButtonStyleTuple = type_1.tuple("default", "primary", "danger");
var ButtonSizeTuple = type_1.tuple("small", "default", "large");
var ButtonHTMLTypes = type_1.tuple("submit", "button", "reset");
exports.Button = function (props) {
    var btnType = props.btnType, loading = props.loading, theme = props.theme;
    var finalTheme = __assign(__assign({}, colors_1.default), theme);
    return loading ? (React.createElement(exports.ButtonDefault, { btnType: btnType, theme: finalTheme, disabled: true }, "Loading...")) : btnType ? (getButtonType(__assign(__assign({}, props), { theme: finalTheme }), btnType)) : null;
};
var getButtonType = function (props, type) {
    var _a = props.btnType, btnType = _a === void 0 ? "outlined" : _a, _b = props.btnStyle, btnStyle = _b === void 0 ? "default" : _b, icon = props.icon, title = props.title, _c = props.onClick, onClick = _c === void 0 ? function () { } : _c, _d = props.href, href = _d === void 0 ? "#" : _d, _e = props.target, target = _e === void 0 ? "blank" : _e, _f = props.htmlType, htmlType = _f === void 0 ? "button" : _f, rest = __rest(props, ["btnType", "btnStyle", "icon", "title", "onClick", "href", "target", "htmlType"]);
    var handleClick = function (e) {
        if (btnType !== "link") {
            onClick(e);
        }
    };
    switch (type) {
        case "solid":
            return (React.createElement(ButtonSolid, __assign({ onClick: handleClick, btnType: btnType, btnStyle: btnStyle, type: htmlType }, rest),
                icon ? getIcon(icon, btnStyle, btnType, rest) : null,
                getTitle(icon, title)));
        case "link":
            return (React.createElement(exports.Link, __assign({ href: href, target: target, btnStyle: btnStyle, onClick: handleClick }, rest),
                icon ? getIcon(icon, btnStyle, btnType, rest) : null,
                getTitle(icon, title)));
        default:
            return (React.createElement(exports.ButtonDefault, __assign({ onClick: handleClick, btnType: btnType, btnStyle: btnStyle, type: htmlType }, rest),
                icon ? getIcon(icon, btnStyle, btnType, rest) : null,
                getTitle(icon, title)));
    }
};
var getIcon = function (icon, style, btnType, rest) {
    var size = rest.size, theme = rest.theme;
    var dimensions = 16;
    if (size === "large") {
        dimensions = 18;
    }
    else if (size === "small") {
        dimensions = 14;
    }
    if (btnType === "outlined" || btnType === "link") {
        switch (style) {
            case "primary":
                return (React.createElement(icon_1.default, { name: icon, color: theme.PRIMARY, size: dimensions }));
            case "danger":
                return (React.createElement(icon_1.default, { name: icon, color: theme.DANGER, size: dimensions }));
            default:
                return (React.createElement(icon_1.default, { name: icon, color: theme.GREY2, size: dimensions }));
        }
    }
    return React.createElement(icon_1.default, { name: icon, color: theme.WHITE, size: dimensions });
};
var getTitle = function (icon, title) { return (React.createElement("span", { style: { marginLeft: icon && title ? "7px" : "0px" } }, title)); };
var BaseButton = styled_components_1.css(templateObject_5 || (templateObject_5 = __makeTemplateObject(["\n\t", ";\n\tcursor: pointer;\n\tpadding: 12px 15px;\n\tdisplay: inline-flex;\n\talign-items: flex-end;\n\tjustify-content: center;\n\tborder-radius: 4px;\n\t-webkit-user-select: none;\n\t-moz-user-select: none;\n\t-ms-user-select: none;\n\tuser-select: none;\n\ttouch-action: manipulation;\n\tfont-size: 16px;\n\tline-height: 16px;\n\tmin-height: 40px;\n\ttransition: color 0.3s ease-in;\n\t&:focus {\n\t\toutline: 0;\n\t}\n\t", ";\n\t", "\n\t", ";\n\t", ";\n"], ["\n\t", ";\n\tcursor: pointer;\n\tpadding: 12px 15px;\n\tdisplay: inline-flex;\n\talign-items: flex-end;\n\tjustify-content: center;\n\tborder-radius: 4px;\n\t-webkit-user-select: none;\n\t-moz-user-select: none;\n\t-ms-user-select: none;\n\tuser-select: none;\n\ttouch-action: manipulation;\n\tfont-size: 16px;\n\tline-height: 16px;\n\tmin-height: 40px;\n\ttransition: color 0.3s ease-in;\n\t&:focus {\n\t\toutline: 0;\n\t}\n\t",
    ";\n\t",
    "\n\t",
    ";\n\t",
    ";\n"])), app_1.GlobalStyles, function (_a) {
    var block = _a.block;
    return block && styled_components_1.css(templateObject_1 || (templateObject_1 = __makeTemplateObject(["\n\t\t\tmin-width: 100%;\n\t\t"], ["\n\t\t\tmin-width: 100%;\n\t\t"])));
}, function (_a) {
    var disabled = _a.disabled;
    return disabled && styled_components_1.css(templateObject_2 || (templateObject_2 = __makeTemplateObject(["\n\t\t\tpointer-events: none;\n\t\t"], ["\n\t\t\tpointer-events: none;\n\t\t"])));
}, function (_a) {
    var size = _a.size;
    return size === "small" && styled_components_1.css(templateObject_3 || (templateObject_3 = __makeTemplateObject(["\n\t\t\tfont-size: 14px;\n\t\t\tline-height: 14px;\n\t\t\tmin-height: 36px;\n\t\t\tpadding: 8px 15px;\n\t\t"], ["\n\t\t\tfont-size: 14px;\n\t\t\tline-height: 14px;\n\t\t\tmin-height: 36px;\n\t\t\tpadding: 8px 15px;\n\t\t"])));
}, function (_a) {
    var size = _a.size;
    return size === "large" && styled_components_1.css(templateObject_4 || (templateObject_4 = __makeTemplateObject(["\n\t\t\tfont-size: 18px;\n\t\t\tline-height: 18px;\n\t\t\tmin-height: 48px;\n\t\t\tpadding: 14px 15px;\n\t\t"], ["\n\t\t\tfont-size: 18px;\n\t\t\tline-height: 18px;\n\t\t\tmin-height: 48px;\n\t\t\tpadding: 14px 15px;\n\t\t"])));
});
exports.ButtonDefault = styled_components_1.default.button(templateObject_9 || (templateObject_9 = __makeTemplateObject(["\n\t", ";\n\tcolor: ", ";\n\tborder: 1px solid ", ";\n\tbackground-color: transparent;\n\t&:hover {\n\t\tbackground-color: ", ";\n\t}\n\t&:active {\n\t\tcolor: ", ";\n\t\tborder-color: ", ";\n\t}\n\t@media screen and (max-width: 767px) {\n\t\t&:hover {\n\t\t\tbackground-color: inherit !important;\n\t\t}\n\t}\n\t", ";\n\t", ";\n\n\t", ";\n"], ["\n\t", ";\n\tcolor: ", ";\n\tborder: 1px solid ", ";\n\tbackground-color: transparent;\n\t&:hover {\n\t\tbackground-color: ", ";\n\t}\n\t&:active {\n\t\tcolor: ", ";\n\t\tborder-color: ", ";\n\t}\n\t@media screen and (max-width: 767px) {\n\t\t&:hover {\n\t\t\tbackground-color: inherit !important;\n\t\t}\n\t}\n\t",
    ";\n\t",
    ";\n\n\t",
    ";\n"])), BaseButton, function (_a) {
    var theme = _a.theme;
    return theme.GREY1;
}, function (_a) {
    var theme = _a.theme;
    return theme.GREY4;
}, function (_a) {
    var theme = _a.theme;
    return theme.BACKGROUND_GREY;
}, function (_a) {
    var theme = _a.theme;
    return theme.BLACK;
}, function (_a) {
    var theme = _a.theme;
    return theme.GREY3;
}, function (_a) {
    var btnStyle = _a.btnStyle, theme = _a.theme;
    return btnStyle === "primary" && styled_components_1.css(templateObject_6 || (templateObject_6 = __makeTemplateObject(["\n\t\t\tcolor: ", "\n\t\t\tborder-color: ", "\n\t\t\t&:hover {\n\t\t\t\tcolor: ", "\n\t\t\t\tborder-color: ", "\n\t\t\t\tbackground-color: ", ";\n\t\t\t}\n\t\t\t&:active {\n\t\t\t\tcolor: ", "\n\t\t\t\tborder-color: ", "\n\t\t\t}\n\t\t"], ["\n\t\t\tcolor: ", "\n\t\t\tborder-color: ", "\n\t\t\t&:hover {\n\t\t\t\tcolor: ", "\n\t\t\t\tborder-color: ", "\n\t\t\t\tbackground-color: ", ";\n\t\t\t}\n\t\t\t&:active {\n\t\t\t\tcolor: ", "\n\t\t\t\tborder-color: ", "\n\t\t\t}\n\t\t"])), theme.PRIMARY, theme.PRIMARY, theme.PRIMARY_LIGHT, theme.PRIMARY_LIGHT, polished_1.transparentize(0.9, theme.PRIMARY_LIGHT), theme.PRIMARY_DARK, theme.PRIMARY_DARK);
}, function (_a) {
    var btnStyle = _a.btnStyle, theme = _a.theme;
    return btnStyle === "danger" && styled_components_1.css(templateObject_7 || (templateObject_7 = __makeTemplateObject(["\n\t\t\tborder-color: ", ";\n\t\t\tcolor: ", ";\n\t\t\t&:hover {\n\t\t\t\tcolor: ", ";\n\t\t\t\tborder-color: ", ";\n\t\t\t\tbackground-color: ", ";\n\t\t\t}\n\t\t\t&:active {\n\t\t\t\tcolor: ", ";\n\t\t\t\tborder-color: ", ";\n\t\t\t}\n\t\t"], ["\n\t\t\tborder-color: ", ";\n\t\t\tcolor: ", ";\n\t\t\t&:hover {\n\t\t\t\tcolor: ", ";\n\t\t\t\tborder-color: ", ";\n\t\t\t\tbackground-color: ", ";\n\t\t\t}\n\t\t\t&:active {\n\t\t\t\tcolor: ", ";\n\t\t\t\tborder-color: ", ";\n\t\t\t}\n\t\t"])), theme.DANGER, theme.DANGER, theme.DANGER_LIGHT, theme.DANGER_LIGHT, polished_1.transparentize(0.9, theme.DANGER_LIGHT), theme.DANGER_DARK, theme.DANGER_DARK);
}, function (_a) {
    var disabled = _a.disabled, theme = _a.theme;
    return disabled && styled_components_1.css(templateObject_8 || (templateObject_8 = __makeTemplateObject(["\n\t\t\tcolor: ", ";\n\t\t\tborder-color: ", ";\n\t\t"], ["\n\t\t\tcolor: ", ";\n\t\t\tborder-color: ", ";\n\t\t"])), theme.DISABLED, theme.DISABLED);
});
var ButtonSolid = styled_components_1.default.button(templateObject_14 || (templateObject_14 = __makeTemplateObject(["\n\t", ";\n\tcolor: ", ";\n\tborder: 1px solid transparent;\n\tbackground-color: ", ";\n\t", "\n\t", ";\n\t", ";\n\n\t", ";\n"], ["\n\t", ";\n\tcolor: ", ";\n\tborder: 1px solid transparent;\n\tbackground-color: ", ";\n\t",
    "\n\t",
    ";\n\t",
    ";\n\n\t",
    ";\n"])), BaseButton, function (_a) {
    var theme = _a.theme;
    return theme.WHITE;
}, function (_a) {
    var theme = _a.theme;
    return theme.GREY3;
}, function (_a) {
    var theme = _a.theme;
    return styled_components_1.css(templateObject_10 || (templateObject_10 = __makeTemplateObject(["\n\t\t\t@media (hover: hover) and (pointer: fine) {\n\t\t\t\t&:hover {\n\t\t\t\t\tbackground-color: ", ";\n\t\t\t\t}\n\t\t\t}\n\t\t\t&:active {\n\t\t\t\tbackground-color: ", ";\n\t\t\t}\n\t\t"], ["\n\t\t\t@media (hover: hover) and (pointer: fine) {\n\t\t\t\t&:hover {\n\t\t\t\t\tbackground-color: ", ";\n\t\t\t\t}\n\t\t\t}\n\t\t\t&:active {\n\t\t\t\tbackground-color: ", ";\n\t\t\t}\n\t\t"])), polished_1.transparentize(0.3, theme.GREY3), theme.GREY2);
}, function (_a) {
    var btnStyle = _a.btnStyle, theme = _a.theme;
    return btnStyle === "primary" && styled_components_1.css(templateObject_11 || (templateObject_11 = __makeTemplateObject(["\n\t\t\tbackground-color: ", ";\n\t\t\t@media (hover: hover) and (pointer: fine) {\n\t\t\t\t&:hover {\n\t\t\t\t\tbackground-color: ", ";\n\t\t\t\t}\n\t\t\t}\n\t\t\t&:active {\n\t\t\t\tbackground-color: ", ";\n\t\t\t}\n\t\t"], ["\n\t\t\tbackground-color: ", ";\n\t\t\t@media (hover: hover) and (pointer: fine) {\n\t\t\t\t&:hover {\n\t\t\t\t\tbackground-color: ", ";\n\t\t\t\t}\n\t\t\t}\n\t\t\t&:active {\n\t\t\t\tbackground-color: ", ";\n\t\t\t}\n\t\t"])), theme.PRIMARY, polished_1.transparentize(0.2, theme.PRIMARY), theme.PRIMARY_DARK);
}, function (_a) {
    var btnStyle = _a.btnStyle, theme = _a.theme;
    return btnStyle === "danger" && styled_components_1.css(templateObject_12 || (templateObject_12 = __makeTemplateObject(["\n\t\t\tbackground-color: ", ";\n\t\t\t&:hover {\n\t\t\t\tbackground-color: ", ";\n\t\t\t}\n\t\t\t&:active {\n\t\t\t\tbackground-color: ", ";\n\t\t\t}\n\t\t"], ["\n\t\t\tbackground-color: ", ";\n\t\t\t&:hover {\n\t\t\t\tbackground-color: ", ";\n\t\t\t}\n\t\t\t&:active {\n\t\t\t\tbackground-color: ", ";\n\t\t\t}\n\t\t"])), theme.DANGER, polished_1.transparentize(0.2, theme.DANGER), theme.DANGER_DARK);
}, function (_a) {
    var disabled = _a.disabled, theme = _a.theme;
    return disabled && styled_components_1.css(templateObject_13 || (templateObject_13 = __makeTemplateObject(["\n\t\t\tbackground-color: ", ";\n\t\t"], ["\n\t\t\tbackground-color: ", ";\n\t\t"])), theme.DISABLED);
});
exports.Link = styled_components_1.default.a(templateObject_19 || (templateObject_19 = __makeTemplateObject(["\n\t", ";\n\tdisplay: inline-flex;\n\talign-items: center;\n\ttext-decoration: none;\n\tbackground: none;\n\tborder: 0;\n\t", "\n\t", ";\n\n\t", "\n\t", "\n"], ["\n\t", ";\n\tdisplay: inline-flex;\n\talign-items: center;\n\ttext-decoration: none;\n\tbackground: none;\n\tborder: 0;\n\t",
    "\n\t",
    ";\n\n\t",
    "\n\t",
    "\n"])), BaseButton, function (_a) {
    var theme = _a.theme;
    return styled_components_1.css(templateObject_15 || (templateObject_15 = __makeTemplateObject(["\n\t\t\tcolor: ", ";\n\t\t\t&:hover {\n\t\t\t\tbackground: ", ";\n\t\t\t}\n\t\t\t&:active {\n\t\t\t\tcolor: ", ";\n\t\t\t}\n\t\t"], ["\n\t\t\tcolor: ", ";\n\t\t\t&:hover {\n\t\t\t\tbackground: ", ";\n\t\t\t}\n\t\t\t&:active {\n\t\t\t\tcolor: ", ";\n\t\t\t}\n\t\t"])), theme.GREY1 || colors_1.default.GREY1, theme.BACKGROUND_GREY || colors_1.default.BACKGROUND_GREY, theme.BLACK || colors_1.default.BLACK);
}, function (_a) {
    var disabled = _a.disabled, theme = _a.theme;
    return disabled && styled_components_1.css(templateObject_16 || (templateObject_16 = __makeTemplateObject(["\n\t\t\tcolor: ", ";\n\t\t"], ["\n\t\t\tcolor: ", ";\n\t\t"])), theme.DISABLED || colors_1.default.DISABLED);
}, function (_a) {
    var btnStyle = _a.btnStyle, theme = _a.theme;
    return btnStyle === "primary" && styled_components_1.css(templateObject_17 || (templateObject_17 = __makeTemplateObject(["\n\t\t\tcolor: ", ";\n\t\t\t&:hover {\n\t\t\t\tcolor: ", ";\n\t\t\t\tbackground: ", ";\n\t\t\t}\n\t\t\t&:active {\n\t\t\t\tcolor: ", ";\n\t\t\t}\n\t\t"], ["\n\t\t\tcolor: ", ";\n\t\t\t&:hover {\n\t\t\t\tcolor: ", ";\n\t\t\t\tbackground: ",
        ";\n\t\t\t}\n\t\t\t&:active {\n\t\t\t\tcolor: ", ";\n\t\t\t}\n\t\t"])), theme.PRIMARY || colors_1.default.PRIMARY, theme.PRIMARY_LIGHT || colors_1.default.PRIMARY_LIGHT, polished_1.transparentize(0.9, theme.PRIMARY || colors_1.default.PRIMARY), theme.PRIMARY_LIGHT || colors_1.default.PRIMARY_LIGHT);
}, function (_a) {
    var btnStyle = _a.btnStyle, theme = _a.theme;
    return btnStyle === "danger" && styled_components_1.css(templateObject_18 || (templateObject_18 = __makeTemplateObject(["\n\t\t\tcolor: ", ";\n\t\t\t&:hover {\n\t\t\t\tcolor: ", ";\n\t\t\t\tbackground: ", ";\n\t\t\t}\n\t\t\t&:active {\n\t\t\t\tcolor: ", ";\n\t\t\t}\n\t\t"], ["\n\t\t\tcolor: ", ";\n\t\t\t&:hover {\n\t\t\t\tcolor: ", ";\n\t\t\t\tbackground: ",
        ";\n\t\t\t}\n\t\t\t&:active {\n\t\t\t\tcolor: ", ";\n\t\t\t}\n\t\t"])), theme.DANGER || colors_1.default.DANGER, theme.DANGER_LIGHT || colors_1.default.DANGER_LIGHT, polished_1.transparentize(0.9, theme.DANGER || colors_1.default.DANGER), theme.DANGER_DARK || colors_1.default.DANGER_DARK);
});
exports.Button.propTypes = {
    /** Display type of button from "outlined", "solid", "link"  */
    btnType: prop_types_1.default.oneOf(ButtonTypeTuple),
    /** Display style of button from "default", "primary", "danger" */
    btnStyle: prop_types_1.default.oneOf(ButtonStyleTuple),
    disabled: prop_types_1.default.bool,
    /** Size options - "small", "default", "large" */
    size: prop_types_1.default.oneOf(ButtonSizeTuple),
    /** Text of the button to display */
    title: prop_types_1.default.string,
    /** Loading state of the button -*/
    loading: prop_types_1.default.bool,
    /** Icon name or component to be displayed */
    icon: prop_types_1.default.oneOfType([prop_types_1.default.string, prop_types_1.default.element]),
    /** Set the button to full width with block={true} */
    block: prop_types_1.default.bool
};
exports.Button.defaultProps = {
    btnType: "outlined",
    btnStyle: "default",
    disabled: false,
    size: "default",
    loading: false,
    block: false,
    theme: colors_1.default
};
exports.default = React.memo(styled_components_1.withTheme(exports.Button));
var templateObject_1, templateObject_2, templateObject_3, templateObject_4, templateObject_5, templateObject_6, templateObject_7, templateObject_8, templateObject_9, templateObject_10, templateObject_11, templateObject_12, templateObject_13, templateObject_14, templateObject_15, templateObject_16, templateObject_17, templateObject_18, templateObject_19;
