// @turtlemint/mint-ui/dist/src/range-selector/index.js
"use strict";
var __makeTemplateObject = (this && this.__makeTemplateObject) || function (cooked, raw) {
    if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
    return cooked;
};
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.RangeSelector = void 0;
var React = __importStar(require("react"));
var styled_components_1 = __importDefault(require("styled-components"));
var react_range_1 = require("react-range");
var colors_1 = __importDefault(require("../__utils/colors"));
var app_1 = require("../app");
exports.RangeSelector = function (_a) {
    var _b = _a.min, min = _b === void 0 ? 0 : _b, _c = _a.max, max = _c === void 0 ? 10000 : _c, _d = _a.step, step = _d === void 0 ? 100 : _d, _e = _a.value, value = _e === void 0 ? 0 : _e, _f = _a.disabled, disabled = _f === void 0 ? false : _f, labelFormatter = _a.labelFormatter, _g = _a.trackFilledColor, trackFilledColor = _g === void 0 ? colors_1.default.PRIMARY : _g, onChange = _a.onChange, onFinalChange = _a.onFinalChange, rest = __rest(_a, ["min", "max", "step", "value", "disabled", "labelFormatter", "trackFilledColor", "onChange", "onFinalChange"]);
    var handleChange = function (values) {
        onChange(values[0]);
    };
    var handleFinalChange = function (values) {
        onFinalChange && onFinalChange(values[0]);
    };
    return (React.createElement(RangeSelectorWrapper, null,
        React.createElement(react_range_1.Range, { disabled: disabled, step: step, min: min, max: max, values: [value], onChange: handleChange, onFinalChange: handleFinalChange, renderTrack: function (_a) {
                var props = _a.props, children = _a.children;
                return (React.createElement(TrackWrapper, { onMouseDown: props.onMouseDown, onTouchStart: props.onTouchStart, style: __assign({}, props.style) }, React.createElement(TrackLine, { ref: props.ref, min: min, max: max, value: value, disabled: disabled, trackFilledColor: trackFilledColor }, children)));
            }, renderThumb: function (_a) {
                var props = _a.props;
                return (React.createElement(StyledThumb /* key={props.key}  */, __assign({}, props, { disabled: true }), rest.showLabel ? (React.createElement(StyledThumbLabel, null, labelFormatter
                    ? labelFormatter(value.toString())
                    : value.toString())) : null));
            } }),
        React.createElement("div", { style: { position: "relative" } },
            rest.startLabel ? (React.createElement(StartLabel, null, rest.startLabel)) : null,
            rest.endLabel ? React.createElement(EndLabel, null, rest.endLabel) : null)));
};
var RangeSelectorWrapper = styled_components_1.default.div(templateObject_1 || (templateObject_1 = __makeTemplateObject(["\n\t", ";\n\tpadding: 30px;\n"], ["\n\t", ";\n\tpadding: 30px;\n"])), app_1.GlobalStyles);
var StyledThumbLabel = styled_components_1.default.div(templateObject_2 || (templateObject_2 = __makeTemplateObject(["\n\t", ";\n\tbottom: 25px;\n\tpadding: 1px 3px;\n\tcolor: ", ";\n\tposition: absolute;\n\tdisplay: inline-block;\n\twhite-space: nowrap;\n\tbackground-color: ", ";\n\tpadding: 6px;\n\tleft: -10px;\n\tborder-radius: 9px;\n\tfont-size: 11px;\n"], ["\n\t", ";\n\tbottom: 25px;\n\tpadding: 1px 3px;\n\tcolor: ", ";\n\tposition: absolute;\n\tdisplay: inline-block;\n\twhite-space: nowrap;\n\tbackground-color: ", ";\n\tpadding: 6px;\n\tleft: -10px;\n\tborder-radius: 9px;\n\tfont-size: 11px;\n"])), app_1.GlobalStyles, colors_1.default.GREY1, colors_1.default.GREY4);
var StartLabel = styled_components_1.default.div(templateObject_3 || (templateObject_3 = __makeTemplateObject(["\n\tposition: absolute;\n\tbackground-color: ", ";\n\tpadding: 5px;\n\tfont-size: 10px;\n\tborder-radius: 3px;\n\tfloat: left;\n"], ["\n\tposition: absolute;\n\tbackground-color: ", ";\n\tpadding: 5px;\n\tfont-size: 10px;\n\tborder-radius: 3px;\n\tfloat: left;\n"])), colors_1.default.GREY4);
var EndLabel = styled_components_1.default.div(templateObject_4 || (templateObject_4 = __makeTemplateObject(["\n\tposition: absolute;\n\tright: 0;\n\tbackground-color: ", ";\n\tpadding: 5px;\n\tfont-size: 10px;\n\tborder-radius: 3px;\n\tfloat: right;\n"], ["\n\tposition: absolute;\n\tright: 0;\n\tbackground-color: ", ";\n\tpadding: 5px;\n\tfont-size: 10px;\n\tborder-radius: 3px;\n\tfloat: right;\n"])), colors_1.default.GREY4);
var TrackWrapper = styled_components_1.default.div(templateObject_5 || (templateObject_5 = __makeTemplateObject(["\n\theight: 36px;\n\tdisplay: flex;\n\twidth: 100%;\n\tposition: relative;\n"], ["\n\theight: 36px;\n\tdisplay: flex;\n\twidth: 100%;\n\tposition: relative;\n"])));
var TrackLine = styled_components_1.default.div(templateObject_6 || (templateObject_6 = __makeTemplateObject(["\n\theight: 8px;\n\twidth: 100%;\n\tborder-radius: 4px;\n\tbackground: ", ";\n\talign-self: center;\n"], ["\n\theight: 8px;\n\twidth: 100%;\n\tborder-radius: 4px;\n\tbackground: ",
    ";\n\talign-self: center;\n"])), function (props) {
    return react_range_1.getTrackBackground({
        values: [props.value],
        colors: [
            props.disabled ? colors_1.default.GREY3 : props.trackFilledColor,
            colors_1.default.GREY4
        ],
        min: props.min,
        max: props.max
    });
});
var StyledThumb = styled_components_1.default.div(templateObject_7 || (templateObject_7 = __makeTemplateObject(["\n\twidth: 20px;\n\theight: 20px;\n\tbackground: ", ";\n\tbox-shadow: 0 0px 2px 0 ", ", 0 0px 2px 0 ", ";\n\tz-index: 3;\n\tborder-radius: 16px;\n\tcursor: ", ";\n\tposition: absolute;\n\tdisplay: inline-block;\n\twhite-space: nowrap;\n"], ["\n\twidth: 20px;\n\theight: 20px;\n\tbackground: ", ";\n\tbox-shadow: 0 0px 2px 0 ", ", 0 0px 2px 0 ", ";\n\tz-index: 3;\n\tborder-radius: 16px;\n\tcursor: ", ";\n\tposition: absolute;\n\tdisplay: inline-block;\n\twhite-space: nowrap;\n"])), colors_1.default.WHITE, colors_1.default.GREY2, colors_1.default.GREY2, function (props) { return (props.disabled ? null : "pointer"); });
// RangeSelector.propTypes = {
// 	min: PropTypes.number,
// 	max: PropTypes.number,
// 	step: PropTypes.number,
// 	value: PropTypes.number,
// 	disabled: PropTypes.bool,
// 	onChange: PropTypes.func,
// 	onFinalChange: PropTypes.func,
// 	thumbLabel: PropTypes.string,
// 	showLabel: PropTypes.bool,
// 	startLabel: PropTypes.node,
// 	endLabel: PropTypes.node,
// 	trackFilledColor: PropTypes.string
// };
exports.RangeSelector.defaultProps = {
    min: 0,
    max: 10000,
    step: 100,
    value: 0,
    disabled: false,
    showLabel: true,
    trackFilledColor: colors_1.default.PRIMARY
};
exports.default = exports.RangeSelector;
var templateObject_1, templateObject_2, templateObject_3, templateObject_4, templateObject_5, templateObject_6, templateObject_7;
