// @turtlemint/mint-ui/dist/src/form/form-item.js
"use strict";
var __makeTemplateObject = (this && this.__makeTemplateObject) || function (cooked, raw) {
    if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
    return cooked;
};
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.HelpText = exports.Label = exports.FormItem = void 0;
var React = __importStar(require("react"));
var styled_components_1 = __importStar(require("styled-components"));
var colors_1 = __importDefault(require("../__utils/colors"));
exports.FormItem = function (_a) {
    var _b;
    var _c = _a.name, name = _c === void 0 ? "" : _c, _d = _a.label, label = _d === void 0 ? "" : _d, helpText = _a.helpText, rules = _a.rules, _e = _a.state, state = _e === void 0 ? {} : _e, _f = _a.handleChange, handleChange = _f === void 0 ? function () { } : _f, _g = _a.errors, errors = _g === void 0 ? {} : _g, _h = _a.handleError, handleError = _h === void 0 ? function () { } : _h, _j = _a.layout, layout = _j === void 0 ? "horizontal" : _j, _k = _a.inline, inline = _k === void 0 ? false : _k, labelCol = _a.labelCol, fieldCol = _a.fieldCol, style = _a.style, children = _a.children, bottomSpace = _a.bottomSpace;
    var handleBlur = function (value) {
        rules ? handleError(rules, name, value) : null;
    };
    return (React.createElement(Wrapper, { layout: layout, style: style, bottomSpace: bottomSpace, className: "formItemWrapper" },
        layout !== "inline" ? (React.createElement(exports.Label, { inline: inline, layout: layout, error: errors[name], labelCol: labelCol }, label)) : null,
        React.createElement(Field, { fieldCol: fieldCol },
            React.cloneElement(children, {
                onChange: handleChange,
                onBlur: handleBlur,
                name: name,
                value: (_b = state[name]) !== null && _b !== void 0 ? _b : children.props.value,
                error: false
            }),
            helpText ? React.createElement(exports.HelpText, null, helpText) : null,
            errors[name] ? React.createElement(ErrorText, null, errors[name]) : null)));
};
var Wrapper = styled_components_1.default.div(templateObject_3 || (templateObject_3 = __makeTemplateObject(["\n\tmargin-bottom: 20px;\n\tdisplay: flex;\n\tflex-wrap: wrap;\n\talign-items: center;\n\tjustify-content: flex-start;\n\t", ";\n\t", ";\n"], ["\n\tmargin-bottom: 20px;\n\tdisplay: flex;\n\tflex-wrap: wrap;\n\talign-items: center;\n\tjustify-content: flex-start;\n\t",
    ";\n\t",
    ";\n"])), function (_a) {
    var layout = _a.layout;
    return layout === "inline" && styled_components_1.css(templateObject_1 || (templateObject_1 = __makeTemplateObject(["\n\t\t\tdisplay: inline-flex;\n\t\t\tmargin-left: 20px;\n\t\t"], ["\n\t\t\tdisplay: inline-flex;\n\t\t\tmargin-left: 20px;\n\t\t"])));
}, function (_a) {
    var bottomSpace = _a.bottomSpace;
    return bottomSpace && styled_components_1.css(templateObject_2 || (templateObject_2 = __makeTemplateObject(["\n\t\t\tmargin-bottom: ", "px;\n\t\t"], ["\n\t\t\tmargin-bottom: ", "px;\n\t\t"])), bottomSpace);
});
var Field = styled_components_1.default.div(templateObject_4 || (templateObject_4 = __makeTemplateObject(["\n\tflex: ", ";\n"], ["\n\tflex: ", ";\n"])), function (_a) {
    var fieldCol = _a.fieldCol;
    return (fieldCol ? fieldCol.span : 4);
});
exports.Label = styled_components_1.default.label(templateObject_9 || (templateObject_9 = __makeTemplateObject(["\n\tflex: ", ";\n\tcolor: ", ";\n\t", ";\n\tfont-weight: 500;\n\tcolor: ", ";\n\t", ";\n\t", ";\n\t", ";\n"], ["\n\tflex: ", ";\n\tcolor: ", ";\n\t",
    ";\n\tfont-weight: 500;\n\tcolor: ", ";\n\t",
    ";\n\t",
    ";\n\t",
    ";\n"])), function (_a) {
    var labelCol = _a.labelCol;
    return (labelCol ? labelCol.span : 1);
}, colors_1.default.GREY1, function (_a) {
    var error = _a.error;
    return error && styled_components_1.css(templateObject_5 || (templateObject_5 = __makeTemplateObject(["\n\t\t\tcolor: ", ";\n\t\t"], ["\n\t\t\tcolor: ", ";\n\t\t"])), colors_1.default.DANGER);
}, function (_a) {
    var error = _a.error;
    return (error ? colors_1.default.DANGER : "initial");
}, function (_a) {
    var layout = _a.layout;
    return layout === "vertical" && styled_components_1.css(templateObject_6 || (templateObject_6 = __makeTemplateObject(["\n\t\t\tmargin-right: 0px;\n\t\t"], ["\n\t\t\tmargin-right: 0px;\n\t\t"])));
}, function (_a) {
    var layout = _a.layout;
    return layout === "inline" && styled_components_1.css(templateObject_7 || (templateObject_7 = __makeTemplateObject(["\n\t\t\tmargin: 0;\n\t\t\tmargin-right: 10px;\n\t\t"], ["\n\t\t\tmargin: 0;\n\t\t\tmargin-right: 10px;\n\t\t"])));
}, function (_a) {
    var layout = _a.layout, inline = _a.inline;
    return layout === "vertical" &&
        !inline && styled_components_1.css(templateObject_8 || (templateObject_8 = __makeTemplateObject(["\n\t\t\tmin-width: 100%;\n\t\t\tmargin-bottom: 10px;\n\t\t"], ["\n\t\t\tmin-width: 100%;\n\t\t\tmargin-bottom: 10px;\n\t\t"])));
});
var ErrorText = styled_components_1.default.div(templateObject_10 || (templateObject_10 = __makeTemplateObject(["\n\tcolor: ", ";\n\tmargin-top: 4px;\n\tword-wrap: break-word;\n\tmax-width: 328px;\n"], ["\n\tcolor: ", ";\n\tmargin-top: 4px;\n\tword-wrap: break-word;\n\tmax-width: 328px;\n"])), colors_1.default.DANGER);
exports.HelpText = styled_components_1.default.p(templateObject_11 || (templateObject_11 = __makeTemplateObject(["\n\tcolor: ", ";\n\tfont-size: 12px;\n\tmargin: 4px 0;\n\tmax-width: 328px;\n"], ["\n\tcolor: ", ";\n\tfont-size: 12px;\n\tmargin: 4px 0;\n\tmax-width: 328px;\n"])), colors_1.default.GREY3);
exports.default = React.memo(exports.FormItem);
var templateObject_1, templateObject_2, templateObject_3, templateObject_4, templateObject_5, templateObject_6, templateObject_7, templateObject_8, templateObject_9, templateObject_10, templateObject_11;
